analysis/3_coregister-and-source-recon-bulk.py

```diff
@@ -0,0 +1,239 @@
+"""Coregistration and source reconstruction 
+Created AXL 28/10/24
+
+The script should be first run for all subjects. Then for subjects
+whose coregistration looked a bit off we re-run this script just for that
+particular subject.
+
+This does not use OSL batch utils/config-file setup.
+"""
+
+############## ------- Import packages and define functions needed ---------- ############
+
+import os
+import os.path as op
+import osl
+from pprint import pprint
+from osl import utils
+from osl import source_recon
+import numpy as np
+import coinsmeg_data as coinsmeg
+from IPython.display import HTML, display
+from glob import glob
+import pathlib
+
+import mne
+from mne import read_forward_solution 
+
+from osl.source_recon.rhino.polhemus import extract_polhemus_from_info
+from osl.source_recon import rhino, beamforming, parcellation # for calculating beamformer weights
+
+# functions
+def copy_polhemus_files(polhemus_dir, recon_dir, subject):
+    polhemus_headshape = np.loadtxt(op.join(polhemus_dir, 'polhemus_headshape.txt'))
+    polhemus_nasion = np.loadtxt(op.join(polhemus_dir, 'polhemus_nasion.txt'))
+    polhemus_rpa = np.loadtxt(op.join(polhemus_dir, 'polhemus_rpa.txt'))
+    polhemus_lpa = np.loadtxt(op.join(polhemus_dir, 'polhemus_lpa.txt'))
+    
+    #  Get coreg filenames
+    filenames = source_recon.rhino.get_coreg_filenames(recon_dir, subject)
+
+    # Save
+    np.savetxt(filenames["polhemus_nasion_file"], polhemus_nasion)
+    np.savetxt(filenames["polhemus_rpa_file"], polhemus_rpa)
+    np.savetxt(filenames["polhemus_lpa_file"], polhemus_lpa)
+    np.savetxt(filenames["polhemus_headshape_file"], polhemus_headshape)
+
+############## ------- Directories ---------- ############
+
+# Directories
+data_dir = coinsmeg.RAW_DIR # this is the same as BASE_DIR as raw data is stored in the base directory
+preproc_dir = coinsmeg.PREPROCESSED_DIR
+recon_dir = op.join(coinsmeg.DERIVATIVES_DIR, "recon")
+
+# Setting of which parcellation file to use
+parcellation_version = 'HarvOxf-sub-Schaefer100-combined-2mm_4d_ds8'
```

Comment:
Could you make the parcellation an argument of the script that can be passed through the command-line (using argparse, for example)? This is so that we can run the source reconstruction for different parcellations without having to modify the code of the script.

This could also be done in a follow-up issue & pull request, if you prefer.

Reply:
I realize that you are running this script using the VScode interface, and that the command line may not be as convenient for you, so feel free to put the above comment aside for now.

Reply:
Ah I see what you mean now. Fix added with argparse – now we can use as e.g.,
```
python analysis/3_coregister-and-source-recon-bulk.py --parcellation_version Glasser52_binary_space-MNI152NLin6_res-1x1x1
```